Source/FieldSolver/ImplicitSolvers/ImplicitSolver.cpp

```diff
@@ -89,6 +91,324 @@ Array<LinOpBCType,AMREX_SPACEDIM> ImplicitSolver::convertFieldBCToLinOpBC (const
     return lbc;
 }
 
+void ImplicitSolver::SaveEandJ ()
+{
+
+    // Copy Efield_fp and current_fp to Efield_fp_save and current_fp_save
+    // Do this BEFORE call to SyncCurrentAndRho() so that BCs are set for J
+
+    using warpx::fields::FieldType;
+    for (int lev = 0; lev < m_num_amr_levels; ++lev) {
+        ablastr::fields::VectorField E = m_WarpX->m_fields.get_alldirs(FieldType::Efield_fp, lev);
+        ablastr::fields::VectorField E0 = m_WarpX->m_fields.get_alldirs(FieldType::Efield_fp_save, lev);
+        amrex::MultiFab::Copy(*E0[0], *E[0], 0, 0, 1, E[0]->nGrowVect());
```

Comment:
For completeness, this could be `amrex::MultiFab::Copy(*E0[0], *E[0], 0, 0, E[0].nComp(), E[0]->nGrowVect());`